leap_c/ocp/acados/utils/create_solver.py

```diff
@@ -0,0 +1,328 @@
+            terminal stage.
+        n_batch_max: Maximum batch size.
+        num_threads: Number of threads used in the batch solver.
+    """
+    ocp.solver_options.with_batch_functionality = True
+
+    # translate cost terms to external to allow
+    # implicit differentiation for a p_global parameter.
+    if ocp.cost.cost_type_0 not in ["EXTERNAL", None]:
+        ocp.translate_initial_cost_term_to_external(
+            cost_hessian=ocp.solver_options.hessian_approx
+        )
+    if ocp.cost.cost_type not in ["EXTERNAL"]:
+        ocp.translate_intermediate_cost_term_to_external(
+            cost_hessian=ocp.solver_options.hessian_approx
+        )
+    if ocp.cost.cost_type_e not in ["EXTERNAL"]:
+        ocp.translate_terminal_cost_term_to_external(
+            cost_hessian=ocp.solver_options.hessian_approx
+        )
+
+    _turn_on_warmstart(ocp)
+
+    if export_directory is None:
+        export_directory = Path(mkdtemp())
+        add_delete_hook = True
+    else:
+        export_directory = Path(export_directory)
+        add_delete_hook = False
+
+    ocp.code_export_directory = str(export_directory / "c_generated_code")
+    json_file = str(export_directory / "acados_ocp.json")
+
+    try:
+        batch_solver = AcadosOcpBatchSolver(
+            ocp,
+            json_file=json_file,
+            N_batch_max=n_batch_max,
+            num_threads_in_batch_solve=num_threads,
+            build=False,
+            generate=False,
+        )
+    except FileNotFoundError:
+        batch_solver = AcadosOcpBatchSolver(
+            ocp,
+            json_file=json_file,
+            N_batch_max=n_batch_max,
+            num_threads_in_batch_solve=num_threads,
+            build=True,
+        )
+
+    if discount_factor is not None:
+        _set_discount_factor(batch_solver, discount_factor)
+
+    if add_delete_hook:
+        DeleteDirectoryHook(batch_solver, export_directory)
+
+    return batch_solver
+
+
+def create_forward_backward_batch_solvers(
+    ocp: AcadosOcp,
+    sensitivity_ocp: AcadosOcp | None = None,  # type:ignore
+    export_directory: str | Path | None = None,
+    discount_factor: float | None = None,
+    n_batch_max: int = 256,
+    num_threads: int = 4,
+):
+    # check if we can use the forward solver for the backward pass.
+    need_backward_solver = _check_need_sensitivity_solver(ocp)
+
+    if need_backward_solver:
+        ocp.solver_options.with_solution_sens_wrt_params = True
+        ocp.solver_options.with_value_sens_wrt_params = True
+
+    forward_batch_solver = create_batch_solver(
+        ocp,
+        export_directory=export_directory,
+        discount_factor=discount_factor,
+        n_batch_max=n_batch_max,
+        num_threads=num_threads,
+    )
+
+    if not need_backward_solver:
+        return forward_batch_solver, forward_batch_solver
+
+    if sensitivity_ocp is None:
+        sensitivity_ocp: AcadosOcp = deepcopy(
```

Comment:
Probably merge this (and following lines) with `derive_sensitivity_ocp` (there are some settings that are being set which are not being set in `derive_sensitivty_ocp`) and then use `derive_sensitivity_ocp`?